example/lib/go_router_example.dart

```diff
@@ -0,0 +1,345 @@
+          StatefulShellBranch(
+            routes: <RouteBase>[
+              GoRoute(
+                // The screen to display as the root in the third tab of the
+                // bottom navigation bar.
+                path: '/c',
+                builder:
+                    (final BuildContext context, final GoRouterState state) =>
+                        const RootScreen(
+                  label: 'C',
+                  detailsPath: '/c/details',
+                ),
+                routes: <RouteBase>[
+                  GoRoute(
+                    path: 'details',
+                    builder: (
+                      final BuildContext context,
+                      final GoRouterState state,
+                    ) =>
+                        DetailsScreen(
+                      label: 'C',
+                      extra: state.extra,
+                    ),
+                  ),
+                ],
+              ),
+            ],
+          ),
+        ],
+      ),
+    ],
+  );
+
+  @override
+  Widget build(final BuildContext context) => MaterialApp.router(
+        title: 'Flutter Demo',
+        theme: ThemeData(
+          colorSchemeSeed: Colors.deepPurple,
+          useMaterial3: true,
+        ),
+        routerConfig: _router,
+      );
+}
+
+/// Builds the "shell" for the app by building a Scaffold with a
+/// BottomNavigationBar, where [child] is placed in the body of the Scaffold.
+class AdaptiveNavigationShell extends StatelessWidget {
+  /// Constructs an [AdaptiveNavigationShell].
+  AdaptiveNavigationShell({required this.navigationShell, super.key});
+
+  /// The navigation shell and container for the branch Navigators.
+  final StatefulNavigationShell navigationShell;
+
+  final List<AdaptiveDestination> destinations = <AdaptiveDestination>[
+    const AdaptiveDestination(
+      initialPath: '/a',
+      icon: Icon(Icons.home_outlined),
+      selectedIcon: Icon(Icons.home),
+      label: 'A',
+    ),
+    const AdaptiveDestination(
+      initialPath: '/b',
+      icon: Icon(Icons.feed_outlined),
+      selectedIcon: Icon(Icons.feed),
+      label: 'B',
+    ),
+    const AdaptiveDestination(
+      initialPath: '/c',
+      icon: Icon(Icons.person_outlined),
+      selectedIcon: Icon(Icons.person),
+      label: 'C',
+    ),
+  ];
+
+  @override
+  Widget build(final BuildContext context) => AdaptiveNavigation(
+        destinations: destinations,
+        navigationTypeResolver: (final BuildContext context) {
+          final double screenWidth = MediaQuery.of(context).size.width;
+          if (screenWidth > 700) {
+            return NavigationType.rail;
+          } else {
+            return NavigationType.bottom;
+          }
+        },
+        onDestinationChanged: (
+          final BuildContext context,
+          final String initialPath,
+          final int index,
+        ) =>
+            navigationShell.goBranch(index),
+        initialIndex: navigationShell.currentIndex,
```

Comment:
This could be moved to above the ``onDestinationChanged`` so that it is not separating the navigation-related methods.

Reply:
done